sdk/openai/Azure.AI.OpenAI/src/Custom/ChatCompletionsOptions.cs

```diff
@@ -15,28 +16,28 @@ namespace Azure.AI.OpenAI
     public partial class ChatCompletionsOptions
     {
         /// <inheritdoc cref="CompletionsOptions.ChoicesPerPrompt"/>
-        [CodeGenMember("N")]
         public int? ChoicesPerPrompt { get; set; }
 
         /// <inheritdoc cref="CompletionsOptions.FrequencyPenalty"/>
         public float? FrequencyPenalty { get; set; }
 
         /// <inheritdoc cref="CompletionsOptions.TokenSelectionBiases"/>
-        public IDictionary<int, int> TokenSelectionBiases { get; }
+        public IDictionary<int, int> TokenSelectionBiases
+        {
+            get
```

Comment:
Per https://github.com/Azure/azure-sdk-for-net/issues/35096, we have a convention that at least *some* (List) collections in input models are (root, container) read-only, with contents written into them. That is, `Messages = new List<ChatMessage>() {...}` would be inconsistent (and also nonstandard for emission purposes).

If it's not unduly burdensome, I'd like to have consistency with collections in the input model since it seems confusing to not allow `Messages = new ...` but then allow `TokenSelectionBiases = new ...` in the same input model object.

Dictionary initializer syntax still works nicely, too, so e.g. this is possible:
```
var requestOptions = new CompletionsOptions()
{
    Prompts = { "What's the weather like in Seattle?" },
    TokenSelectionBiases =
    {
        [3201] = -50, // don't say "rain" as much
    },
};
```

Reply:
Oh, I apologize for missing the real problem here so blatantly! Alas, you're 100% right and my attempt to be clever and avoid the `Record<>` custom deserialization problem overlooked the most obvious issue -- namely that writing to the collection would actually end up in the request.

I went back to full custom serialization for the options classes (which gets rid of the awkward auto-conversion) and also added a bit of test coverage to ensure that the field is actually exercised.